src/util/loopUnrolling.ml

```diff
@@ -446,9 +446,6 @@ class loopUnrollingVisitor(func) = object
           annotateArrays b;
           match s.skind with
           | Loop (b,loc, loc2, break , continue) ->
-            (* We copy the statement to later be able to modify it without worrying about invariants *)
-            let s = { s with sid = s.sid } in
```

Comment:
Why is this now unnecessary? To the contrary, in https://github.com/goblint/analyzer/pull/563#discussion_r789751464 it was deemed necessary.

Reply:
Yes, it was, because it covered up some other bugs. After those are now fixed, it is correct to keep one copy of the statement identical: Since loops are unrolled bottom-up, first the innermost loop gets duplicated (with the last instance remaining the same). Then, the outer loop is unrolled, maintaining the invariant that there is at most one copy.